### DISCUSSION TITLE
Thread pinning with PostgreSQL JDBC driver

### DISCUSSION THREAD

**Discussion by davecramer** [quarkusio/quarkus#33325] [Community]
This https://quarkus.io/guides/virtual-threads#pinning-cases should now be historical with the release of 42.6.0 of the PostgreSQL JDBC driver. Virtually all synchronized methods have been replaced with reentrant lock. See https://jdbc.postgresql.org/changelogs/2023-03-17-42.6.0-release/ for details.
It would be great if you could update your site to reflect the current state.

**Comment by maxandersen** [comment]
@cescoffier ^^

**Reply by cescoffier** [reply]
I need to run more tests. Yes, this version is far better. My last run was still seeing a bit of pinning, however I was using hibernate and agroal (so might not be because of postgresql).

I also need to check if there is any thread local used for polling. As we now know, it causes large performance issues.

**Comment by davecramer** [comment]
@cescoffier if it is still pinning because of the driver we want to know!
